pkg/kapp/clusterapply/add_or_update_change.go

```diff
@@ -295,7 +311,17 @@ type UpdatePlainStrategy struct {
 func (c UpdatePlainStrategy) Op() ClusterChangeApplyStrategyOp { return updateStrategyPlainAnnValue }
 
 func (c UpdatePlainStrategy) Apply() error {
-	updatedRes, err := c.aou.identifiedResources.Update(c.newRes)
+	var updatedRes ctlres.Resource
+	var err error
+
+	if c.aou.opts.ServerSideApply {
+		updatedRes, err = ctlres.WithIdentityAnnotation(c.newRes, func(r ctlres.Resource) (ctlres.Resource, error) {
```

Comment:
it seems that we have two types of Patch operations: one where we are working with a resource, and one where we are working with some "minimal/raw" patch. instead of exposing WithIdentityAnnotation outside of IdentifiedResources resources class, lets keep existing Patch method to work with Resource class (instead of []byte) and change it so that it can add annotation on the fly. lets also add PatchRaw which does take []byte. this should result in keeping WithIdentityAnnotation as private.